File: pyg_1_3/pyg_parent/pyg_manager_web/src/main/webapp/js/service/brandService.js
// 自定义服务
// 参数1：服务名称
// 参数2：function
app.service("brandService", function ($http) {
    // 查询品牌列表
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    };
    // 分页
    this.findPage = function (page, rows) {
        return $http.get('../brand/findPage.do?page=' + page + '&rows=' + rows);
    };
    // 回显
    this.findBrand = function (id) {
        return $http.get("../brand/findBrand.do?id="+id);
    };
    // 批量删除
    this.dele = function (ids) {
        return $http.get("../brand/delete.do?ids="+ids);
    };
    // 分页条件查询
    this.search = function (page, rows, entity) {
        return $http.post("../brand/search.do?page="+page+"&rows="+rows, entity);
    };
    // 添加品牌
    this.add = function (entity) {
        return $http.post("../brand/add.do", entity);
    };
    // 修改品牌
    this.update = function (entity) {
        return $http.post("../brand/update.do", entity);
    };
    // 选择选项列表
    this.selectOptionList = function () {
        return $http.get('../brand/selectOptionList.do');
    }
});

File: pyg_1_3/pyg_parent/pyg_sellergoods_interface/src/main/java/com/pyg/sellergoods/service/BrandService.java
package com.pyg.sellergoods.service;

import com.pyg.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 作者：杨立波
 * 时间：2018/8/29 20:17
 */
public interface BrandService {

    /**
     * 查询所有品牌
     *
     * @return 品牌列表
     */
    public List<TbBrand> findAll();

    /**
     * 查询分页结果
     *
     * @param pageNum  页码
     * @param pageSize 记录数
     * @return 分页结果
     */
    public PageResult findPage(int pageNum, int pageSize);

    /**
     * 添加品牌
     *
     * @param brand 品牌对象
     */
    public void add(TbBrand brand);

    /**
     * 查询指定品牌
     *
     * @param id 品牌id
     * @return 品牌对象
     */
    public TbBrand findBrand(Long id);

    /**
     * 修改品牌
     *
     * @param brand 品牌对象
     */
    public void update(TbBrand brand);

    /**
     * 删除品牌
     *
     * @param ids 品牌id数组
     */
    public void delete(Long[] ids);

    /**
     * 按条件查询分页
     *
     * @param brand    品牌对象
     * @param pageNum  页码
     * @param pageSize 记录数
     * @return 分页结果
     */
    public PageResult findPage(TbBrand brand, int pageNum, int pageSize);

    /**
     * 选择规格列表
     *
     * @return 规格列表
     */
    public List<Map> selectOptionList();
}
